### PR TITLE
COL-744, asset-library-search gets new filter: by Section

### PR DESCRIPTION
**Pull request by johncrossman**
https://jira.ets.berkeley.edu/jira/browse/COL-744

List of distinct section names can be extracted from the list `users` already available in angular directive. SQL behind search now has `array_position(u.canvas_course_sections, :section)`

### CODE REVIEW

**Comment by johncrossman** [comment]
image

File: node_modules/col-assets/lib/api.js [hunk 1]
```diff
@@ -367,6 +371,10 @@ var getAssets = module.exports.getAssets = function(ctx, filters, sort, limit, o
     sqlQuery += ' AND au.user_id = :user';
   }
 
+  if (filters.section) {
+    sqlQuery += ' AND (array_position(u.canvas_course_sections, :section) > 0)';
+  }
+
```
Comment (johncrossman) [inline node_modules/col-assets/lib/api.js]: Postgres doc: https://www.postgresql.org/docs/9.5/static/functions-array.html

**Comment by paulkerschen** [comment]
Thanks, @johncrossman, code looks good to me. I hate to bring this up since our test framework is so difficult to work with, but what do you think about adding some coverage for the back-end component in `col-assets/tests/test-assets.js`? We have a utility method set up to verify search results, so the grunt work would involve creating test users and associating them with different sections.

**Comment by johncrossman** [comment]
@paulkerschen I accept that challenge. It'll be another climb up the learning curve.

**Comment by johncrossman** [comment]
I'm closing this PR in favor of upcoming PR w/ test coverage. Please stand by.

**Pull request closed without merging**